packages/dashboard/src/app/views/myStories/content/storiesView/index.js

```diff
@@ -174,12 +176,29 @@ function StoriesView({
       menuItemActions: {
         default: () => setContextMenuId(-1),
         [STORY_CONTEXT_MENU_ACTIONS.COPY_STORY_LINK]: handleCopyStoryLink,
-        [STORY_CONTEXT_MENU_ACTIONS.DELETE]: handleDeleteStory,
-        [STORY_CONTEXT_MENU_ACTIONS.DUPLICATE]: handleDuplicateStory,
+        ...(apiCallbacks?.trashStory && {
+          [STORY_CONTEXT_MENU_ACTIONS.DELETE]: handleDeleteStory,
+        }),
+        ...(apiCallbacks?.duplicateStory && {
+          [STORY_CONTEXT_MENU_ACTIONS.DUPLICATE]: handleDuplicateStory,
+        }),
         [STORY_CONTEXT_MENU_ACTIONS.OPEN_STORY_LINK]: handleOpenStoryInEditor,
-        [STORY_CONTEXT_MENU_ACTIONS.RENAME]: handleRenameStory,
+        ...(apiCallbacks?.updateStory && {
+          [STORY_CONTEXT_MENU_ACTIONS.RENAME]: handleRenameStory,
+        }),
       },
-      menuItems: STORY_CONTEXT_MENU_ITEMS,
+      menuItems: STORY_CONTEXT_MENU_ITEMS.filter((menu) => {
+        switch (menu.value) {
+          case STORY_CONTEXT_MENU_ACTIONS.DELETE:
+            return typeof apiCallbacks?.trashStory === 'function';
+          case STORY_CONTEXT_MENU_ACTIONS.DUPLICATE:
+            return typeof apiCallbacks?.duplicateStory === 'function';
+          case STORY_CONTEXT_MENU_ACTIONS.RENAME:
+            return typeof apiCallbacks?.updateStory === 'function';
+          default:
+            return true;
+        }
+      }),
```

Comment:
Do we really need this?

Instead of checking for the API callbacks here, should we just rely on the capabilities being set?

Menu items are filtered based on user capabilities here:

https://github.com/google/web-stories-wp/blob/7849181a9f4ba9a5b56a9123bcb3dc845c9cda42/packages/dashboard/src/components/popoverMenu/story-menu-generator.js#L48-L50

Reply:
Yes, good point, "Duplicate" story actually didn't have any capability assigned.

 
![image](https://user-images.githubusercontent.com/6297436/143848494-f5c62838-626a-43a1-8659-95811fac7dc8.png)


Removed this filter and added `hasEditAction` capability in https://github.com/google/web-stories-wp/pull/9752/commits/1621c6ae519652c8babaed335ea1ea9110bcd83f